packages/core/src/common/selection-command-handler.ts

```diff
@@ -28,17 +28,27 @@ export class SelectionCommandHandler<S> implements CommandHandler {
 
     execute(...args: any[]): Object | undefined {
         const selection = this.getSelection(...args);
-        return selection ? (this.options.execute as any)(selection, ...args) : undefined;
+        return selection ? (this.options.execute as any)(selection, ...args) : (this.options.execute as any)(undefined);
     }
 
     isVisible(...args: any[]): boolean {
         const selection = this.getSelection(...args);
-        return !!selection && (!this.options.isVisible || (this.options.isVisible as any)(selection as any, ...args));
+        //custom isVisible function overrides default checking mechanism
+        if (!this.options.isVisible) {
+            return !!selection;
+        } else {
+            return (this.options.isVisible as any)(selection as any, ...args);
+        }
     }
 
     isEnabled(...args: any[]): boolean {
         const selection = this.getSelection(...args);
-        return !!selection && (!this.options.isEnabled || (this.options.isEnabled as any)(selection as any, ...args));
+        //custom isEnabled function overrides default checking mechanism
+        if (!this.options.isEnabled) {
```

Comment:
Can you elaborate why the `isVisible` and `isEnabled` code needs to be modified?

Reply:
the main reason is that, with the old implementation, a menu item is grayed-out or hidden if `selection` is undefined, but our goal is exactly to deal with the case when the user tries to upload a file without selecting any directory.

So the modification is to let `options.isEnabled` have a higher priority: if we force `options.isEnabled` to be true, then the menu is enabled even when selection is undefined